src/libasr/asdl_cpp.py

```diff
@@ -2154,11 +2154,11 @@ def visitModule(self, mod):
         self.emit("/" + "*"*78 + "/")
         self.emit("// Deserialization Visitor base class")
         self.emit("")
-        self.emit("template <class Struct>")
-        self.emit("class DeserializationBaseVisitor : public BaseVisitor<Struct>")
+        self.emit("template <class StructType>")
+        self.emit("class DeserializationBaseVisitor : public BaseVisitor<StructType>")
         self.emit("{")
         self.emit("private:")
-        self.emit(  "Struct& self() { return static_cast<Struct&>(*this); }", 1)
+        self.emit(  "StructType& self() { return static_cast<StructType&>(*this); }", 1)
```

Comment:
I think all the changes in this file needs to be reverted?

Reply:
Could you please tell me why this is wrong? We have this same change in LFortran now and I would fix this there too.

Reply:
`Struct` is type name in the template, this file is used to generate the asr.h file. So I think this file doesn't require the change.